[feat] 인증인가 및 관리자 계정 자동 생성, 회원가입 구현

cjn404 opened the pull request:
## #️⃣ Issue Number<!--- ex) #이슈번호, #이슈번호 --> 

- Closes #19  

## 📝 요약(Summary)<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [ ] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)## 💬 공유사항 to 리뷰어<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->
<img width="883" height="316" alt="스크린샷 2025-10-19 오전 12 33 07" src="https://github.com/user-attachments/assets/465c4030-3992-43d0-a9ae-a7de600e5252" />

<img width="893" height="328" alt="스크린샷 2025-10-19 오전 12 34 17" src="https://github.com/user-attachments/assets/c6a6063d-e4ae-42ce-9b27-13d9dc3a3346" />

<img width="843" height="758" alt="스크린샷 2025-10-19 오전 12 34 49" src="https://github.com/user-attachments/assets/cccac499-8306-4a73-8fbc-b9a4d01e167c" />

<img width="876" height="764" alt="스크린샷 2025-10-19 오전 12 35 14" src="https://github.com/user-attachments/assets/6aa4607c-fda7-4b26-b13d-aa8fd250de18" />

<img width="956" height="759" alt="스크린샷 2025-10-19 오전 12 35 38" src="https://github.com/user-attachments/assets/0e8090cc-29e9-4c31-850f-9f2261eb9462" />

<img width="893" height="760" alt="스크린샷 2025-10-19 오전 12 35 55" src="https://github.com/user-attachments/assets/e0fbe32b-95b2-4125-83d8-75567bd3f5c1" />

<img width="822" height="765" alt="스크린샷 2025-10-19 오전 12 36 09" src="https://github.com/user-attachments/assets/2f67f325-acd9-4557-b145-16fbf814111a" />

<img width="853" height="767" alt="스크린샷 2025-10-19 오전 12 36 30" src="https://github.com/user-attachments/assets/d1c322ce-e0c2-474a-80de-5379373beda9" />

<img width="1044" height="754" alt="스크린샷 2025-10-19 오전 12 37 00" src="https://github.com/user-attachments/assets/0a3f1a1d-d373-4b0d-b2ac-3e42408a277f" />

<img width="943" height="763" alt="스크린샷 2025-10-19 오전 12 37 48" src="https://github.com/user-attachments/assets/95502496-9adf-4b3e-8093-3944ce2fd318" />


## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

zerone1202 committed:
[init] closet: Closet 엔티티 연관관계 매핑 및 생성자 수정

- @Builder(access = PROTECTED) 적용으로 생성자 접근 제어
- Closet ↔ ClosetClothesLink 연관관계 설정 및 매핑 정리

zerone1202 committed:
[init] clothes: Clothes 엔티티 연관관계 매핑

- Clothes ↔ ClosetClothesLink 연관관계 설정 및 매핑 정리

zerone1202 committed:
[init] closetclotheslink: ClosetClothesLink 엔티티 연관관계 매핑

- Clothes ↔ ClosetClothesLink ↔ Closet 연관관계 매핑

zerone1202 committed:
[init] closetclotheslink: ClosetClothesLinkRepository 레포지토리 세팅

zerone1202 committed:
[init] closetclotheslink: @Builder 패턴, 정적팩토리메서드 적용

zerone1202 committed:
[init] closet: 개행 정리

cjn404 committed:
[refactor] user: user 엔티티의 컬럼 옵션 수정

- 가독성 있게 옵션 순서 수정
- 디폴트 값도 작성
- 주석 수정

AllSungho committed:
[init] chatParticipatingUser : 초기 세팅

- User, Chatroom 헬퍼 메서드 구현

AllSungho committed:
[init] chatParticipatingUser : 초기 세팅 추가

AllSungho committed:
[init] chatParticipatingUser : 초기 세팅 추가

cjn404 committed:
[feat] auth: AuthSignupRequest 구현 및 user 엔티티 수정

- AuthSignupRequest에 맞춰 user 엔티티 수정

cjn404 committed:
[feat] auth: AuthLoginRequest 구현

cjn404 committed:
[feat] user: user 엔티티 전화번호 UK 처리

Byeongsu-cmd committed:
[feat] category : 카테고리 엔티티 연관관계 셀프 조인으로 재구현

Byeongsu-cmd committed:
[feat] category : 카테고리 초기 데이터 주입 구현

- common에 init 패키지 추가
- init 패키지에 CategoryDataInitializer를 추가하여 시스템 실행 시 카테고리 초기 데이터 주입
- 재실행 시 데이터가 로컬에 존재하는 지 검증
- CategoryCommandServiceImpl에 카테고리 초기 데이터 삽입 메서드 구현 (추후에 개선할 필요가 있음!)

Byeongsu-cmd committed:
[init] successCode: 성공 코드 초기 세팅

Byeongsu-cmd committed:
[feat] category: 카테고리 등록 API

- 카테고리 요청 값 제약
- 카테고리 POST 구현

Byeongsu-cmd committed:
[refactor] Category: 불필요한 로직 제거

Byeongsu-cmd committed:
[feat] Category: 카테고리에러코드 중복 코드 제거

cjn404 committed:
[chore] all: auto-assign yml 파일 추가

- assign 자동 기입
- reviewer 1명 랜덤 지정
- 기존 yml 삭제

cjn404 committed:
[fix] all: auto-assign yml 파일 주석 들여쓰기 수정

cjn404 committed:
[feat] auth: 스프링 시큐리티 의존성 추가

cjn404 committed:
[init] auth: service 인터페이스, 구현체 파일 생성

cjn404 committed:
[docs] all: yml 주석 수정

cjn404 committed:
[feat] auth: AuthUser 구현

cjn404 committed:
[feat] auth: JwtAuthenticationFilter 구현

cjn404 committed:
[feat] auth: JwtAuthenticationToken 구현

cjn404 committed:
[feat] auth: JwtUtil 구현

cjn404 committed:
[feat] auth: SecurityConfig 구현

cjn404 committed:
[refactor] auth: 변수명 수정(role -> userRole)

cjn404 committed:
[feat] user: service 인터페이스, 구현체 파일 생성

cjn404 committed:
[feat] user: 관리자 계정 시드 데이터로 생성

AllSungho committed:
[fix] common : 공통 응답 및 전역 예외 처리 수정

AllSungho committed:
[fix] common : 공통 응답 및 전역 예외 처리 추가 수정

Byeongsu-cmd committed:
[fix] Category: 카테고리 성공코드, 에러코드 수정

Byeongsu-cmd committed:
[init] clothes: 옷 성공코드, 에러코드 초기 세팅

cjn404 committed:
[chore] all: auto_assign.yml 파일 삭제

zerone1202 committed:
[feat] closet: Closet 엔티티 필드 및 로직 수정

- visibility 필드를 isPublic으로 변경 (명확한 네이밍)
 - Builder 및 정적 팩토리 메서드 파라미터 변경
- User 연관관계 추가 (추후 사용 예정)
 - @ManyToOne으로 User 엔티티와 연관관계 매핑
 - FetchType.LAZY (지연 로딩)
- userId 필드 추가 (임시)
 - User 생성 전으로 인한 임시 추가
 - Long 타입으로 user_id를 직접 저장
 - Builder 및 정적 팩토리 메서드 userId 파라미터 추가
- 주석 추가로 필드 설명

zerone1202 committed:
[feat] user: User 엔티티 필드 및 로직 수정

- Closet 연관관계 추가 (추후 사용 예정)
 - @OneToMany으로 Closet 엔티티와 연관관계 매핑

zerone1202 committed:
[feat] closet: ClosetSaveRequest DTO 생성 및 기존 ClosetRequest 클래스명 변경

- ClosetRequest -> ClosetSaveRequest 클래스명 변경
- record 타입으로 불변 객체 구현
- 유효성 검증 어노테이션 적용
 - @NotNull: userId, isPublic 필수 값 검증
 - @NotBlank: name 필수 값 및 공백 검증
 - @Size: name(최대 100자), description(최대 255자) 길이 제한

zerone1202 committed:
[feat] closet: ClosetSaveResponse DTO 생성 및 기존 ClosetResponse 클래스명 변경

- ClosetResponse -> ClosetSaveResponse 클래스명 변경
- record 타입으로 불변 객체 구현
- Builder 패턴 적용
- closet.getUserId()로 userId 조회(User 엔티티 미사용)

Byeongsu-cmd committed:
[refactor] category: 카테고리 코드 리팩토링

zerone1202 committed:
[feat] closet: ClosetCommandService 인터페이스 생성

- ClosetService -> ClosetCommandService 클래스명 및 인터페이스 변경
- createCloset() 메서드 선언
- Command와 Query 분리 원칙 적용 (CQRS)

Byeongsu-cmd committed:
[refactor] clothes: 옷관련 이넘 필드 리팩토링

- size 어노테이션과 이넘의 size가 중복되어 그것을 방지하기 위해 이넘 클래스명 변경
- clothesSize 이넘 값 FREE 추가

zerone1202 committed:
[feat] closet: ClosetCommandServiceImpl 구현체 생성

- ClosetCommandService 인터페이스 구현
- @Transactional 적용
- Closet.create() 정적 팩토리 메서드로 엔티티 생성
- ClosetRepository.save()로 DB 저장
- ClosetSaveResponse.from()으로 응답 DTO 변환
- User 검증 로직 제외 (User 도메인 미완성)

Byeongsu-cmd committed:
[feat] clothes: 옷 엔티티와 디티오 구현

- 유저 값 주석처리 (유저는 추후에 연관관계 맺어서 구현할 예정)

zerone1202 committed:
[feat] closet: ClosetController 생성 및 옷장 등록 API 구현

- POST /api/v1/closets 엔드포인트 추가
- 옷장 등록 기능 구현
- @Valid로 요청 데이터 유효성 검증
- ClosetSuccessCode.CLOSET_CREATED를 활용한 201 Created 응답
- ApiResponse 공통 응답 형식 적용
- 요청: ClosetSaveRequest
  - userId(임시), name, description, imageUrl, isPublic
- 응답: ClosetSaveResponse
  - closetId, userId, name, description, imageUrl, isPublic, createdAt, updatedAt
- 실패 응답:
  - 400 Bad Request: 필수 값 누락, 유효하지 않은 데이터 형식 (@Valid 자동 처리)

zerone1202 committed:
[feat] closet: ClosetErrorCode 생성

- Closet 도메인의 에러 응답 코드 정의
- ErrorCode 인터페이스 구현

zerone1202 committed:
[feat] closet: ClosetSuccessCode 생성

- Closet 도메인의 성공 응답 코드 정의
- SuccessCode 인터페이스 구현
- CLOSET_CREATED: 옷장 등록 성공 (201 Created)

Byeongsu-cmd committed:
[feat] clothes: 옷 등록 Api 구현

- 카테고리 쿼리 서비스를 참조하여 카테고리 데이터 값 호출

zerone1202 committed:
[feat] closet: ClosetException 커스텀 예외 클래스 생성

- GlobalException을 상속받아 Closet 도메인 전용 예외 클래스 구현
- 도메인별 예외 계층 구조 확립

zerone1202 committed:
[feat] closet: Closet 엔티티 내 연관관계 주석 처리

-User 연관관계 현재 미사용으로 주석 처리

zerone1202 committed:
[feat] user: User 엔티티 내 연관관계 주석 처리

- Closet 연관관계 현재 미사용으로 주석 처리

zerone1202 committed:
[feat] closet: ClosetController URL 수정

- yml 파일에 /api 설정되어 있어 URL 수정

zerone1202 committed:
[feat] closet: ClosetSaveRequest 유효성 검증 로직 추가

- 유효성 검증을 위한 @Size 어노테이션 추가
 - 이미지 URL (최대 500자)

zerone1202 committed:
[feat] closet: ClosetSaveResponse 불필요한 빌더 패턴 제거

cjn404 committed:
[feat] auth: deprecated된 메서드 수정

cjn404 committed:
[feat] auth: payload claim을 email -> loginID로 변경

cjn404 committed:
[feat] auth: payload subject를 UUID로 변경 및 블랙리스트 관리용 jti 추가

cjn404 committed:
[feat] auth: 암호화 알고리즘 방식 명시적으로 특정

cjn404 committed:
[feat] auth: auth 경로 인증인가 화이트리스트 처리

cjn404 committed:
[feat] auth: 회원가입 구현

cjn404 committed:
[feat] auth: AuthErrorCode 구현

cjn404 committed:
[feat] auth: AuthSuccessCode 구현

cjn404 committed:
[feat] auth: AuthException 구현

cjn404 committed:
[feat] user: 빌더의 타 도메인 사용을 위해 캡슐화 해제

cjn404 committed:
[feat] auth: AuthResponse 클래스 삭제

cjn404 committed:
[refactor] user: 서비스 Command, Query로 분리

cjn404 committed:
[refactor] auth: 서비스 Command로 구분해서 리네이밍

cjn404 committed:
[refactor] auth: 관리자 계정 자동 생성 관련 user -> auth로 도메인 이동

cjn404 committed:
Merge branch 'dev' into feat/auth-19

cjn404 commented:
CI 실패 원인은 "환경 변수 ${JWT_SECRET_KEY}가 비어 있거나 너무 짧음"이라고 합니다
ci.yml 파일 수정 및 GitHub setting 내 시크릿키 설정 완료입니다

cjn404 committed:
[init] all: 시크릿키 전역 환경 변수로 변경

cjn404 committed:
[init] all: test 환경에서 캐시 비활성화

- CI에서 실패 원인: GitHub Actions 환경에서 Redis에 연결 불가

cjn404 committed:
[init] all: Redis 관련 삭제

cjn404 committed:
[init] all: H2 버전 명시

cjn404 committed:
[init] all: application-test.yml 명시적 활성화

cjn404 committed:
[init] all: application-test.yml 내 시크릿 키 수정

cjn404 committed:
[init] all: application-test.yml&ci.yml 내 시크릿 키 부분 수정

zerone1202 commented on .github/workflows/ci.yml:
```diff
@@ -22,6 +23,4 @@ jobs:
-          JWT_SECRET_KEY: ${{ secrets.JWT_SECRET_KEY }}
-        run: ./gradlew clean check -Dspring.profiles.active=test
\ No newline at end of file
+        run: JWT_SECRET_KEY=${{ secrets.JWT_SECRET_KEY }} ./gradlew clean check -Dspring.profiles.active=test
```
GitHub Actions에서는 env: 블록을 사용하는 것이 더 안정적이고 가독성도 좋은 것으로 알고있습니다!
특히 secrets를 사용할 때는 env: 방식이 보안과 호환성 면에서 더 안전하다고 하더라구요~~!

혹시 env를 run으로 합치신 이유가 있으실까요?

cjn404 replied:
로컬 테스트와 무관하게 CI테스트에서 실패했고 그 원인이 될 수 있다고 해서 바꿔봤습니다~

cjn404 committed:
[feat] auth: 토큰 서브젝트 수정

zerone1202 commented on src/main/java/org/example/ootoutfitoftoday/domain/user/service/query/UserQueryServiceImpl.java:
```diff
@@ -0,0 +1,40 @@
+
+@Service
+@RequiredArgsConstructor
+@Transactional
```
QueryService가 조회만 관련된 서비스라 @Transactional(readOnly = true)로 수정해야 할 것 같습니당 :)

또한, 코드 컨벤션에서 클래스 단위로 @Transactional을 붙이기로 하여서 
아래 메서드에 붙은 @Transactional(readOnly = true)들은 제거해야 할 것 같습니다!

cjn404 replied:
수정 완료입니다~

zerone1202 commented on src/main/java/org/example/ootoutfitoftoday/security/config/SecurityConfig.java:
```diff
@@ -0,0 +1,57 @@
+                .rememberMe(AbstractHttpConfigurer::disable)     // 서버가 쿠키 발급하여 자동 로그인
+
+                .authorizeHttpRequests(auth -> auth
+                        .requestMatchers("/auth/signup", "/auth/login").permitAll()
```
비회원도 볼 수 있는 화면들이 존재하기에, 저희 API명세서 참고해서 URL 추가 해야 할 것 같습니다 !
+ URL에 버전이 빠졌네용

cjn404 replied:
수정 완료입니다~

zerone1202 commented on src/main/java/org/example/ootoutfitoftoday/security/filter/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,133 @@
+        String requestUri = httpRequest.getRequestURI();
+
+        // 인증 불필요 경로는 필터 스킵
+        if (requestUri.startsWith("/api/auth/")) {
```
버전 추가가 필요합니다~
-> "/api/v1/auth/"

cjn404 replied:
수정 완료입니다~

zerone1202 commented on src/main/java/org/example/ootoutfitoftoday/security/filter/JwtAuthenticationFilter.java:
```diff
@@ -0,0 +1,133 @@
+        // 인증 불필요 경로는 필터 스킵
+        if (requestUri.startsWith("/api/auth/")) {
+            chain.doFilter(httpRequest, httpResponse);
+            return;
```
return 줄 바꿈이 필요합니당~

cjn404 replied:
수정 완료입니다~

AllSungho commented on src/main/java/org/example/ootoutfitoftoday/domain/auth/service/command/AuthCommandService.java:
```diff
@@ -0,0 +1,17 @@
+import org.example.ootoutfitoftoday.domain.auth.dto.request.AuthSignupRequest;
+
+public interface AuthCommandService {
+
```
해당 인터페이스는 다른 API에서 필요로하는 메서드를 제공하기 위한 Service이므로 수정 부탁드리겠습니다!

cjn404 replied:
넵

cjn404 committed:
[style] user: 컨벤션 통일

cjn404 committed:
[fix] auth: URL 누락 수정

cjn404 committed:
[fix] auth: url 누락 수정

cjn404 committed:
[fix] auth: 미인증 url 누락 수정

AllSungho commented on src/main/java/org/example/ootoutfitoftoday/domain/user/entity/User.java:
```diff
@@ -101,6 +101,26 @@ public static User create(
+            String password,
+            String phoneNumber
+    ) {
+        return User.builder()
```
여깁니다!

cjn404 committed:
[style] user: 코드 스타일 컨벤션에 맞춰 수정

cjn404 committed:
[fix] auth: 메서드별 접근 권한 구분

cjn404 committed:
[fix] auth: url 순서 오류 수정

cjn404 committed:
[feat] auth: 인증 불필요한 경로-필터 스킵하도록 구현

cjn404 committed:
[fix] auth: url에서 context-path 제거

cjn404 committed:
[style] auth: 코드 스타일 컨벤션에 맞춰 수정

# Conflicts:
#	src/main/java/org/example/ootoutfitoftoday/security/filter/JwtAuthenticationFilter.java

AllSungho commented on src/main/java/org/example/ootoutfitoftoday/domain/auth/init/AdminDataInitializer.java:
```diff
@@ -0,0 +1,33 @@
+    private final AuthCommandService authCommandService;
+
+    @Bean
+    public CommandLineRunner initAdmin() {
```
리턴 컨벤션 여기 필요한 부분 있습니다!

cjn404 replied:
수정 완료입니다!

cjn404 committed:
[style] auth: 코드 컨벤션에 맞춰 개행 추가

AllSungho reviewed (approved):
수고 많으셨습니다!

AllSungho reviewed (approved):
확인했습니다!
